src/Visualization/Python/PlotAlongLine.py

```diff
@@ -144,19 +167,56 @@ def plot_along_line_command(
         color="black" if len(vars) == 1 else None,
     )
 
-    # Plot the selected quantities
-    for y, var_name in zip(vars_on_line, vars):
-        plt.plot(
-            x,
-            y,
-            label=var_name,
-            **plot_kwargs,
-        )
-
-    # Configure the axes
+    # Set up the figure
+    fig, ax = plt.figure(), plt.gca()
     plt.xlim(x[0], x[-1])
+    if y_bounds:
+        plt.ylim(*y_bounds)
     plt.xlabel(x_label)
+    # Keep track of plot lines to update in animation
+    lines = [
+        plt.plot([], [], label=var_name, **plot_kwargs)[0] for var_name in vars
+    ]
     plt.legend()
+    time_label = plt.annotate(
+        "",
+        xy=(0, 0),
+        xycoords="axes fraction",
+        xytext=(4, 3),
+        textcoords="offset points",
+        ha="left",
+        va="bottom",
+        fontsize=9,
+    )
+
+    def update_plot(obs_id, obs_time):
+        vars_on_line = interpolate_to_points(
+            h5_files,
+            subfile_name=subfile_name,
+            observation_id=obs_id,
+            tensor_components=vars,
+            target_points=target_coords,
+            num_threads=num_threads,
+        )
+        for y, var_name, line in zip(vars_on_line, vars, lines):
+            line.set_data(x, y)
+        time_label.set_text(f"t = {obs_time:g}")
+        ax.relim()
+        ax.autoscale_view()
```

Comment:
Not a change request, but this could potentially create weird movies if the bounds are changing quickly. We'd have to do something more sophisticated to fix this though.

Reply:
The alternative is looking through all files and observations to determine bounds. I had to do that for PlotSlice so colors remain the same, but here I thought it's better to autoscale and have the user fix bounds if needed.